### DISCUSSION TITLE
Replacing object doesn't update observables.  Gets stuck on older object.

### DISCUSSION THREAD

**Discussion by InputNeuron** [mobxjs/mobx#2899] [General]
Previously I had a version of our app that had an object with just a few strings, numbers, etc.

if I would call setFoo() on a string the UI would reflect that change.

Now I've changed foo to be an object (class instance) and it's Reactive and has makeObservable called but when I replace it the UI doesn't update.

What am I doing wrong here? 

I just updated and I'm running the latest version of mobx.

**Comment by urugator** [comment]
Please provide ideally runnable reproduction.
